### PR TITLE
RI: Show a list of "chips" with interests on the UI

### PR DESCRIPTION
**Pull request by ashiagr**
Master issue: #12027

This PR just gets mocked list of interests from `ReaderTagRepository` and displays them as list of "chips" on the UI. 

### To test:
0. Enable runtime feature flag as described in #12050 and switch to build variant where it is enabled.
1. Go to Reader tab.
2. Wait for a while.
3. Notice that list of "chips" with interests is shown on the UI.

device-2020-06-02-135659

Note:
Still need to refine the "chips" UI (font, colors etc), will do in upcoming PRs.

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt [hunk 1]
```diff
@@ -2,16 +2,19 @@ package org.wordpress.android.ui.reader.discover.interests
...
 import org.wordpress.android.ui.reader.repository.ReaderTagRepository
+import org.wordpress.android.viewmodel.ScopedViewModel
 import javax.inject.Inject
+import javax.inject.Named
 
 class ReaderInterestsViewModel @Inject constructor(
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
     private val readerTagRepository: ReaderTagRepository
-) : ViewModel() {
+) : ScopedViewModel(mainDispatcher) {
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt:17]: Used `ScopedViewModel` here for consistency. Also tried out `viewModelScope` [in this branch](https://github.com/wordpress-mobile/WordPress-Android/tree/issue/12026-ri-show-interests-chips-viewmodelscope)  as our `viewmodel` libraries are upgraded and support it. I'm fine with both approaches. Let me know whatever works :).

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt:17]: I'm fine with both approaches as well ;). Whatever you prefer.

File: WordPress/src/test/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModelTest.kt [hunk 2]
```diff
@@ -14,11 +25,54 @@ class ReaderInterestsViewModelTest {
...
+        assertThat(requireNotNull(viewModel.uiState.value).interests).isEqualTo(mockInterests)
+        assertThat(requireNotNull(viewModel.uiState.value).interestsUiState[0])
+                .isInstanceOf(InterestUiState::class.java)
+        assertThat(requireNotNull(viewModel.uiState.value).interestsUiState[0].title)
+                .isEqualTo(mockInterests[0].tagTitle)
     }
 
     @Test
-    fun `sample test`() {} // TODO:
+    fun `interests ui not updated if no tags received from repo on start`() = test {
```
Comment (malinajirka) [inline WordPress/src/test/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModelTest.kt:54]: I assume we'll remove this test when we introduce loading/error handling, right?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt [hunk 3]
```diff
@@ -1,6 +1,60 @@
...
+        if (initialized) return
+        loadInterests()
+    }
+
+    private fun loadInterests() {
+        launch {
+            val tagList = readerTagRepository.getInterests()
+            if (tagList.isNotEmpty()) {
+                updateUiState(UiState(transformToInterestsUiState(tagList), tagList))
+                if (!initialized) {
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt:33]: I haven't seen this usage of "initialized" anywhere else in the app. I'm not sure it'll still make sense when we introduce loading/error handling. But we'll see :).

Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt:33]: I found `initialized` used in the [ReaderViewModel's start method](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderViewModel.kt#L62).

>  I'm not sure it'll still make sense when we introduce loading/error handling. But we'll see :).

👍

File: WordPress/src/test/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModelTest.kt (hunk 2 above)
Comment (ashiagr) [reply WordPress/src/test/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModelTest.kt:54]: Right.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt (hunk 1 above)
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt]: Swapped `ScopedViewModel` with `viewModelScope` in 29e7481.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt (hunk 3 above)
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt:30]: :D ohh, 🐟 memory. 

👍

**Comment by malinajirka** [comment]
LGTM, thanks

**Pull request merged**